src/components/shared/thread/SelectedActivityGraph.js

```diff
@@ -0,0 +1,350 @@
+      fullThread,
+      filteredThread,
+      callNodeInfo,
+      selectedThreadIndex,
+      changeSelectedCallNode,
+      focusCallTree,
+    } = this.props;
+    const unfilteredStack = fullThread.samples.stack[sampleIndex];
+    if (unfilteredStack === null) {
+      return;
+    }
+
+    const clickedCategory = fullThread.stackTable.category[unfilteredStack];
+    const { callNodeTable, stackIndexToCallNodeIndex } = callNodeInfo;
+    const sampleCallNodes = getSampleCallNodes(
+      filteredThread.samples,
+      stackIndexToCallNodeIndex
+    );
+    const clickedCallNode = sampleCallNodes[sampleIndex];
+    if (clickedCallNode === null) {
+      return;
+    }
+
+    const sampleCategories = getSampleCategories(
+      fullThread.samples,
+      fullThread.stackTable
+    );
+    const chosenCallNode = findBestCallNode(
+      callNodeInfo,
+      sampleCallNodes,
+      sampleCategories,
+      clickedCallNode,
+      clickedCategory
+    );
+    // Change selection twice: First, to clickedCallNode, in order to expand
+    // the whole call path. Then, to chosenCallNode, to get the large-area
+    // graph highlighting.
+    changeSelectedCallNode(
+      selectedThreadIndex,
+      getCallNodePathFromIndex(clickedCallNode, callNodeTable)
+    );
+    changeSelectedCallNode(
+      selectedThreadIndex,
+      getCallNodePathFromIndex(chosenCallNode, callNodeTable)
+    );
+    focusCallTree();
+  }
+  render() {
+    const {
+      fullThread,
+      filteredThread,
+      interval,
+      rangeStart,
+      rangeEnd,
+      callNodeInfo,
+      selectedCallNodeIndex,
+      categories,
+      samplesSelectedStates,
+      treeOrderSampleComparator,
+    } = this.props;
+
+    return (
+      <div>
+        <ThreadActivityGraph
+          interval={interval}
+          fullThread={fullThread}
+          className="selectedThreadActivityGraph"
+          rangeStart={rangeStart}
+          rangeEnd={rangeEnd}
+          onSampleClick={this._onSampleClick}
+          categories={categories}
+          samplesSelectedStates={samplesSelectedStates}
+          treeOrderSampleComparator={treeOrderSampleComparator}
+        />
+        <ThreadStackGraph
+          interval={interval}
+          thread={filteredThread}
+          className="selectedThreadStackGraph"
+          rangeStart={rangeStart}
+          rangeEnd={rangeEnd}
+          callNodeInfo={callNodeInfo}
+          selectedCallNodeIndex={selectedCallNodeIndex}
+          onSampleClick={this._onSampleClick}
+          categories={categories}
+          upsideDown={true}
+        />
+      </div>
+    );
+  }
+}
+
+const SelectedThreadActivityGraphCanvasWithViewport = withChartViewport(
+  SelectedThreadActivityGraphCanvas
+);
+
+function viewportNeedsUpdate() {
+  // By always returning false we prevent the viewport from being
+  // reset and scrolled all the way to the bottom when doing
+  // operations like changing the time selection or applying a
+  // transform.
```

Comment:
Is this comment appropriate / relevant? There's no "all the way to the bottom" in this graph, as far as I know.